code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -706,3 +714,30 @@ GLOBAL_LIST_INIT(wraith_banish_very_short_duration_list, typecacheof(list(
 	banish_check.banish_deactivate()
 	succeed_activate()
 	add_cooldown()
+
+///Return TRUE if we have a block, return FALSE otherwise
+/atom/proc/turf_block_check(atom/target, ignore_can_pass = FALSE, ignore_density = FALSE, ignore_closed_turf = FALSE, ignore_invulnerable = FALSE, ignore_objects = FALSE)
+	var/turf/T = get_turf(target)
```

Comment:
This should not be an atom level proc, it should be on the ability

Reply:
It is called by multiple different abilities; how do you want this done exactly?